src/factorizations/lanczos.jl

```diff
@@ -365,3 +365,326 @@ function lanczosrecurrence(operator, V::OrthonormalBasis, β, orth::ModifiedGram
     end
     return w, α, β
 end
+
+# BlockLanczos
```

Comment:
It's probably useful to put everything below in a separate `factorizations/blocklanczos.jl` file